base/abstractarray.jl

```diff
@@ -340,7 +340,7 @@ function checkbounds(::Type{Bool}, A::AbstractArray, i)
     checkindex(Bool, linearindices(A), i)
 end
 # As a special extension, allow using logical arrays that match the source array exactly
-function checkbounds{_,N}(::Type{Bool}, A::AbstractArray{_,N}, I::AbstractArray{Bool,N})
+function checkbounds(::Type{Bool}, A::AbstractArray{_,N}, I::AbstractArray{Bool,N}) where {_,N}
```

Comment:
would this be better as
function checkbounds(::Type{Bool}, A::AbstractArray{<:Any,N}, I::AbstractArray{Bool,N}) where N
or are those different? @pabloferz